src/util/type.h

```diff
@@ -32,10 +32,18 @@ class typet:public irept
 
   explicit typet(const irep_idt &_id):irept(_id) { }
 
+#if !defined(__GNUC__) || __GNUC__ >= 6
   typet(irep_idt _id, typet _subtype)
     : irept(std::move(_id), {}, {std::move(_subtype)})
   {
   }
+#else
```

Comment:
Note this will also be used under Clang, which pretends to be gcc 4.2.1

Reply:
Ah, thanks a lot, I've added `#defined(__clang__)` to the above `#if`.